python/models/shotgun_sort_filter_proxy_model.py

```diff
@@ -0,0 +1,233 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import sgtk
+
+from sgtk.platform.qt import QtCore, QtGui
+
+shotgun_model = sgtk.platform.import_framework(
+    "tk-framework-shotgunutils",
+    "shotgun_model",
+)
+
+shotgun_globals = sgtk.platform.import_framework(
+    "tk-framework-shotgunutils",
+    "shotgun_globals",
+)
+
+class ShotgunSortFilterProxyModel(QtGui.QSortFilterProxyModel):
+    """
+    A sort/filter proxy model that handles sorting and filtering
+    data in a ShotgunModel by given Shotgun fields on the entities
+    stored therein.
+
+    :ivar filter_by_fields:         A list of string Shotgun field names to filter on.
```

Comment:
No specific reason other than brevity. I'm fine with it being a property instead for the sake of consistency.